purchase_requisition_line_description/models/purchase_requisition_line.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.fields import first
+
+
+class PurchaseRequisitionLine(models.Model):
+    _inherit = "purchase.requisition.line"
+
+    name = fields.Text(string="Product Description")
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        """
+        New method is triggered when the `product_id` field is changed.
+        It updates the `name` field (description) based on the selected product
+        and the partner's language and purchase description.
+        """
+        if self.product_id:
+            partner = (
+                self.requisition_id.purchase_ids.partner_id
+                or first(self.requisition_id.mapped("purchase_ids")).partner_id
+            )
```

Comment:
```suggestion
            partner = first(self.requisition_id.purchase_ids).partner_id
```
Maybe can be simplified.

Reply:
changes are done